### PR TITLE
Fixed an issue with ContentCurator.getContentByProducts not fetching data

### PR DESCRIPTION
**Pull request by Ceiu**
- Worked around an issue in ContentCurator where the getContentByProducts
  method would always return an empty collection

### CODE REVIEW

**Comment by Ceiu** [comment]
**NOTE**: This is a temporary-ish fix for the (silent) issue Hibernate has with querying against embedded entities. Once a better fix is introduced, this commit should be reverted.

**Testing Instructions**:
- Deploy current master with test data
- Import a manifest that contains updates for an existing content set which is used by an existing product that is not being updated in the manifest *and* references a second content set which is *not* referenced in the manifest (I can provide such a manifest on request). This should fail with an exception mentioning a content entity not existing.
- Deploy this branch
- Attempt to import the above manifest again. It should work this time.

File: server/src/main/java/org/candlepin/model/ContentCurator.java [hunk 1]
```diff
@@ -82,16 +86,73 @@ public Content lookupByUuid(String uuid) {
...
+                    if (i != 0) {
+                        builder.append(" OR ");
+                    }
+
+                    builder.append("product_uuid IN (?").append(++i).append(')');
+                }
+
+                builder.append(')');
+
+                SQLQuery query = session.createSQLQuery(builder.toString());
```
Comment (mstead) [inline server/src/main/java/org/candlepin/model/ContentCurator.java]: Due to complex nature of building up this query, lets add a quick debug statement showing the resulting query.

Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/ContentCurator.java]: Done.

File: server/src/main/java/org/candlepin/model/Product.java [hunk 2]
```diff
@@ -185,7 +185,7 @@
     @ElementCollection
     @BatchSize(size = 32)
     @CollectionTable(name = "cp2_product_content", joinColumns = @JoinColumn(name = "product_uuid"))
-    @Column(name = "element")
+    @Column(name = "content_uuid")
     @LazyCollection(LazyCollectionOption.EXTRA) // allows .size() without loading all data
```
Comment (mstead) [inline server/src/main/java/org/candlepin/model/Product.java]: Out of curiosity, what was the purpose of changing this? It seems out of band from this PR. Just looking for some context.

Comment (Ceiu) [inline server/src/main/java/org/candlepin/model/Product.java]: I don't think it has any actual value or affect on anything in the code (I could be wrong though, I'm far from an expert on the annotation side of things). However, the "element" column is just wrong -- it doesn't exist in the cp2_product_content table and is likely just leftover from copy/pasting the annotations from productAttributes above.

Comment (mstead) [inline server/src/main/java/org/candlepin/model/Product.java]: Cool, thanks. 'element' does seem strange. I'd say the @CollectionTable's join column trumps it or something.

**Comment by mstead** [comment]
ACK - Looks good and seems to work. Waiting for tests to complete before merging.

**Pull request merged**